### ISSUE TITLE
[General] Provide javascript and css framework 

### ISSUE THREAD

**Issue by Kaik** [zikula/core#3946]
We need a proper framework for js and css. 
At the moment we use few polyfills, Jquery, translating and routing bootstrap provides some js and that is basically all (requirejs via 3party - > unused)

All js assets are provided via composer -> and composer is not a js package manager!
The solution is #3601 of course but choosing/using js package manager is just the first step, and it is mostly useful for core developers. 

What we want to do (assuming that Zikula wants to be more of a "ready product" rather than, a "bare bone" framework) is to choose one js framework (angular/react etc) + one js templating system and provide that as "Zikula" default choice. That default choice is for 3rd party module developers. 

Of course, any developer can use his own favorite js framework and tools but he will have to build core from the source or in worst case add the 3party library into his module (which is bad!)

Basically, the composer is our manager for PHP and Symfony is our framework, we need something similar for javascript. #3601 is probably the best choice. For js framework well... hard to say ;) js templating I think we could use twigjs as default solution it is not the best but it is compatible with twig and you can nicely reuse twig templates

There is not much to chose with CSS, of course, sass rules... 

to be continued...
